crates/jsonrpc/src/lib.rs

```diff
@@ -22,6 +24,25 @@ pub async fn write_msg(
     Ok(())
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct RPCErr {
+    pub code: i32,
+    pub message: String,
+}
+
+pub async fn write_err(mut out: impl tokio::io::AsyncWrite + Unpin, err: RPCErr) -> Result<()> {
+    use tokio::io::AsyncWriteExt;
+
+    let msg = serde_json::to_string(&err)?;
+
+    let header = format!("Content-Length: {}\r\n\r\n", msg.len());
+    out.write_all(header.as_bytes()).await?;
+    out.write_all(msg.as_bytes()).await?;
+    out.flush().await?;
+
+    Ok(())
+}
```

Comment:
Unrelated, but it was making it difficult to debug errors across RPC boundaries because we weren't propagating them back to the lua layer. This just adds a parallel to `write_msg` that writes a jsonrpc-complaint error response, which is already correctly interpreted by the client side.